dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcRequestHandlerMapping.java

```diff
@@ -42,9 +43,16 @@ protected boolean supportContentType(String contentType) {
 
     @Override
     protected void determineHttpMessageCodec(RequestHandler handler, URL url, HttpRequest request) {
-        HttpMessageCodec codec = CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
-        handler.setHttpMessageDecoder(codec);
-        handler.setHttpMessageEncoder(codec);
+        GrpcCompositeCodec grpcCompositeCodec =
+                (GrpcCompositeCodec) CODEC_FACTORY.createCodec(url, getFrameworkModel(), request.contentType());
+        MethodDescriptor methodDescriptor = DescriptorUtils.findMethodDescriptor(
+                handler.getServiceDescriptor(), handler.getMethodName(), handler.isHasStub());
+        if (methodDescriptor != null) {
+            handler.setMethodDescriptor(methodDescriptor);
+            grpcCompositeCodec.loadPackableMethod(methodDescriptor);
+        }
+        handler.setHttpMessageDecoder(grpcCompositeCodec);
+        handler.setHttpMessageEncoder(grpcCompositeCodec);
```

Comment:
**Security Issue**: The use of findMethodDescriptor without validating the method's name, service descriptor, or the stub flag may lead to the execution of unexpected methods if the input parameters can be influenced externally. <br> **Fix**: Validate the method name, service descriptor, and stub flag before using them to find and set the MethodDescriptor. <br> **Code Suggestion**: 
 ```
 +        if (DescriptorUtils.validateMethodName(handler.getMethodName()) &&
 +            DescriptorUtils.validateServiceDescriptor(handler.getServiceDescriptor()) &&
 +            handler.isHasStub()) {
 +            MethodDescriptor methodDescriptor = DescriptorUtils.findMethodDescriptor(
 +                handler.getServiceDescriptor(), handler.getMethodName(), handler.isHasStub());
 +            if (methodDescriptor != null) {
 +                handler.setMethodDescriptor(methodDescriptor);
 +                grpcCompositeCodec.loadPackableMethod(methodDescriptor);
 +            }
 +        }
 ```